localflavor/gr/forms.py

```diff
@@ -8,12 +8,17 @@ class GRPostalCodeField(RegexValidator):
     """
     Greek Postal code field.
 
-    Format: XXXXX, where X is any digit, and first digit is not 0 or 9.
+    Format: XXX XX or XXXXX, where X is any digit, and first digit is not 0 or 9.
     """
 
     default_error_messages = {
-        'invalid': _('Enter a valid 5-digit postal code.'),
+        "invalid": _(
+            "Enter a valid numeric postal code in the format XXX XX or XXXXX, "
+            "where the first number cannot be 0 or 9."
+        ),
     }
 
     def __init__(self, *args, **kwargs):
-        super(GRPostalCodeField, self).__init__(r'^[12345678]\d{4}$', *args, **kwargs)
+        super(GRPostalCodeField, self).__init__(
+            r"^[1-8]\d{2} \d{2}$|^[1-8]\d{4}$", *args, **kwargs
```

Comment:
I've kept the `[1-8]` here, but the only source I can find for that is [Wikipedia](https://en.wikipedia.org/wiki/Postal_codes_in_Greece). Other regex definitions I've found just use `\d{3} \d{2}|\d{5}`.